lib/ood_core/job/info.rb

```diff
@@ -113,6 +113,21 @@ def initialize(id:, status:, allocated_nodes: [], submit_host: nil,
         @native          = native
       end
 
+      # Create a new Info for a child task
+      # @return [Info] merging the parent and the child task
+      def new_for_child_task(task)
```

Comment:
I'd rename this to `build_child_task`. 

Inspiration is the method naming conventions in Rails for Active Record collection build methods: https://apidock.com/rails/ActiveRecord/Associations/CollectionProxy/build and https://guides.rubyonrails.org/association_basics.html#methods-added-by-belongs-to-build-association-attributes

Reply:
Oh except it isn't a child task that is being built. The equivalent "build" name for new_for_child_task would be something like `build_child_info`